ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -6216,18 +6216,85 @@ IHqlExpression * WorkflowTransformer::transformInternalFunction(IHqlExpression *
 
     HqlExprArray funcdefArgs;
     funcdefArgs.append(*LINK(newBody));
-    unwindChildren(funcdefArgs, newFuncDef, 1);
-    OwnedHqlExpr namedFuncDef = newFuncDef->clone(funcdefArgs);
-    inheritDependencies(namedFuncDef);
 
-    if (ecl->getOperator() == no_embedbody)
+    if (ecl->getOperator() == no_embedbody && ecl->hasAttribute(languageAtom))
+    {
+        IHqlExpression * outofline = newFuncDef->queryChild(0);
+        IHqlExpression * formals = newFuncDef->queryChild(1);
+        IHqlExpression * defaults = newFuncDef->queryChild(2);
+        assertex(outofline->getOperator() == no_outofline);
+        IHqlExpression * bodyCode = outofline->queryChild(0);
+        HqlExprArray attrArgs;
+        ForEachChild(idx, bodyCode)
+        {
+            IHqlExpression *child = bodyCode->queryChild(idx);
+            if (child->isAttribute() && child->queryName() != languageAtom && child->queryName() != importAtom)
+            {
+                StringBuffer attrParam;
+                if (attrArgs.ordinality())
+                    attrParam.append(",");
+                attrParam.append(child->queryName());
+
+                IHqlExpression * value = child->queryChild(0);
+                if (value)
+                    attrParam.append("=");
+                attrArgs.append(*createConstant(attrParam));
+                if (value)
+                    attrArgs.append(*ensureExprType(value, unknownStringType));
+            }
+        }
+        OwnedHqlExpr folded;
+        if (attrArgs.length())
+        {
+            OwnedHqlExpr concat = createUnbalanced(no_concat, unknownStringType, attrArgs);
+            OwnedHqlExpr cast = ensureExprType(concat, unknownVarStringType);
+
+            // It's not legal to use parameters in the options, since it becomes ambiguous whether they should be bound to embed variables or not.
+            // Check that they didn't and give a sensible error message
+            OwnedHqlExpr boundCast = replaceInlineParameters(newFuncDef, cast);
+            if (cast != boundCast)
+                throwError(HQLERR_EmbedParamNotSupportedInOptions);
+
+            folded.setown(foldHqlExpression(cast));
+        }
+        else
+            folded.setown(createConstant(""));
+        HqlExprArray newFormals;
+        HqlExprArray attrs;
```

Comment:
trivial style: I would move this down a line so it is closer to its use - it makes the code slightly easier to read